proposals/solid-oidc-primer/index.bs

```diff
@@ -159,62 +159,78 @@ describes everything the client will need to know to authorize with Alice's spec
 
 Response Body:
 ```json
-{  
-   "issuer":"https://secureauth.example",
-   "authorization_endpoint":"https://secureauth.example/authorize",
-   "token_endpoint":"https://secureauth.example/token",
-   "userinfo_endpoint":"https://secureauth.example/userinfo",
-   "jwks_uri":"https://secureauth.example/jwks",
-   "registration_endpoint":"https://secureauth.example/register",
-   "response_types_supported":[  
-      "code",
-      "code token",
-      "code id_token",
-      "id_token",
-      "id_token token",
-      "code id_token token",
-      "none"
-   ],
-   "response_modes_supported":[  
-      "query",
-      "fragment"
-   ],
-   "grant_types_supported":[  
-      "authorization_code",
-      "implicit",
-      "refresh_token",
-      "client_credentials"
-   ],
-   "subject_types_supported":[  
-      "public"
-   ],
-   "id_token_signing_alg_values_supported":[  
-      "RS256",
-      "RS384",
-      "RS512",
-      "none"
-   ],
-   "token_endpoint_auth_methods_supported":[  
-      "client_secret_basic"
-   ],
-   "token_endpoint_auth_signing_alg_values_supported":[  
-      "RS256"
-   ],
-   "display_values_supported":[  
-
-   ],
-   "claim_types_supported":[  
-      "normal"
-   ],
-   "claims_supported":[  
-
-   ],
-   "claims_parameter_supported":false,
-   "request_parameter_supported":true,
-   "request_uri_parameter_supported":false,
-   "require_request_uri_registration":false,
-   "check_session_iframe":"https://secureauth.example/session",
-   "end_session_endpoint":"https://secureauth.example/logout"
+{
+    "issuer": "https://secureauth.example",
+    "authorization_endpoint": "https://secureauth.example/authorize",
+    "token_endpoint": "https://secureauth.example/token",
+    "userinfo_endpoint": "https://secureauth.example/userinfo",
+    "jwks_uri": "https://secureauth.example/jwks",
+    "registration_endpoint": "https://secureauth.example/register",
+    "response_types_supported": [
+        "code",
+        "code token",
+        "code id_token",
+        "id_token",
+        "id_token token",
```

Comment:
This isn't incorrect, but we shouldn't be recommending implicit flow. I would be inclined to be minimalistic about this and _only_ include the authorization code flow (e.g. `code`) and omit the others.

Reply:
I am not completely clear on this point:
The [OIDC discovery specification](https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderMetadata), states "Dynamic OpenID Providers MUST":
- support response types "code", "id_token", and "token id_token"
- support grant types "authorization_code" and "implicit"

It seems everything but "code" denotes implicit/hybrid flows which [are outdated and have good reasons not to be used](https://oauth.net/2/grant-types/implicit/). Are there edge cases where one might want to use them and are they worthy of the trade-off?

Are `response_types_supported` used to determine anything else than the authorization flow available to clients using url parameter `response_type` at the `authorization_endpoint` URL, for example `https://idp.com/authorization?response_type=code`?

The `grant_types_supported` seem redundant but actually describe what is possible at the `token_endpoint`, meaning if we use `response_type=code` at the authorization endpoint, later in the flow we'll use `grant_type=authorization_code` to get a token. And one might want to support any number of grant types at the token endpoint. I'm not completely clear yet on everything involved in getting token outside of an authorization flow, but I understand "implicit" is the problematic one, is it correct?

So, is it true to state that:
Most Solid OpenID providers will need to support dynamic client registration, which will make them Dynamic OpenID providers, however, one might consider the possibility not to support the implicit flows, which would make them not 100% compliant to the OpenID Connect 1.0 specification but foregoing those redundant and less secure authorization flows as of 2020 represents an interesting trade-off.

Would those authorization flows still be required in a new version of the OpenID specification?

Reply:
Opted for removing all but the "code" response type and advising against the implicit flows.